### ISSUE TITLE
No log file

### ISSUE THREAD

**Issue by fmder** [cdr/code-server#871]
- `code-server` version: development
- OS Version: Ubuntu 16.04.6

## Description

The directories for logging in the user's home directory .cache/code-server/logs are created but the log files are not consistently written. For a lot of our users no log file can be found in the log directories. 

## Steps to Reproduce

When starting an instance of VSCode, whenever we try to access the logs in the output pane, an error File not found like the one pasted bellow are showing up

![image](https://user-images.githubusercontent.com/961474/61896089-31a13080-aee2-11e9-90a2-9d149d0d60f2.png)

**Comment by deansheather** [comment]
You mentioned you were using a development copy. Does this occur for you in the latest release version?

**Comment by antofthy** [comment]
I have see consistent writing (every 5 minutes) while a web browser is connected to a code-server instance to the  file  ".cache/code-server/logs/????/sharedprocess.log"    Where ???? is the highest number.  We clear out all code-server cached files when the user conatiner starts, so there is only even one such sub-directory.

In fact we use this "every 5 minute" connection as part of our determination of if a user is active or not.  We also can tell from this file if code server was started, and not yet used in this file only has 4 lines in it (for user monitoring reports only)

However other than 'activity' the contents of this log file, or any other log file, including normal code-server output (other than for password retrieval),  have not proven to be very useful.

**Comment by deansheather** [comment]
The number in the log directory name is a timestamp.

**Comment by antofthy** [comment]
> The number in the log directory name is a timestamp.
Yes, in UTC - That confused me for a few seconds.  But that does not matter... Just go for the last number, and junk the rest, and you have the current log being updated.

**Comment by fmder** [comment]
With the released version I still have no log

```
$ code-server --version
1.1156-vsc1.33.1
```

My usecase for the log is that  I'm trying to run pytest and somehow it fails. I think the answer why it fails is in the log but I can't find it anywhere. This is the output for the latest folders 

```
$ ls -la ~/.cache/code-server/logs/20190729133123675/output_20190729133203550_20190729T093204/
total 0
drwxr-xr-x 2 rpqjh889 users   6 Jul 29 09:32 .
drwxr-xr-x 6 rpqjh889 users 186 Jul 29 09:32 ..
```

```
$ ls -la ~/.cache/code-server/logs/20190729133123675/exthost20190729133203550/ 
total 0
drwxr-xr-x 3 rpqjh889 users  52 Jul 29 09:32 .
drwxr-xr-x 6 rpqjh889 users 186 Jul 29 09:32 ..
drwxr-xr-x 2 rpqjh889 users  10 Jul 29 09:32 output_logging_20190729T093206
```
```
$ ls -la ~/.cache/code-server/logs/20190729133123675/exthost20190729133203550/output_logging_20190729T093206/
total 0
drwxr-xr-x 2 rpqjh889 users 10 Jul 29 09:32 .
drwxr-xr-x 3 rpqjh889 users 52 Jul 29 09:32 ..
```

**Comment by eVanasse** [comment]
I am facing the same issue. I ran `tree` after running a few Python tests looking for the test output.

![image](https://user-images.githubusercontent.com/24358053/62716433-2d8c0d00-b9d0-11e9-9a3f-eb365db3449d.png)


Note that I'm running it in a Docker container.

**Comment by deansheather** [comment]
Could you please try with the v2 prerelease and see if it's fixed?

**Comment by eVanasse** [comment]
I could not get **code-server** started.

I downloaded the release, copied it in my Docker image, launched the container, entered it then tried to start `code-server`.

Got this error:
![image](https://user-images.githubusercontent.com/24358053/63042198-7db71380-be97-11e9-85e6-c410c85de5d1.png)

I could solve it by installing `ifconfig` and `ip`, but why do we need those with the new release?

**Comment by deansheather** [comment]
Most Linux systems should have ifconfig or ip installed on them anyways by default, it's fairly unusual to see a distro without it. It needs (one of) these programs to determine the mac address of the computer.

Does the log file work in the preview?

**Comment by code-asher** [comment]
VS Code uses the MAC address for the purpose of generating a 
unique machine ID that is used in the telemetry so one solution 
would be to disable telemetry (`--disable-telemetry`).

**Comment by antofthy** [comment]
"`ifconfig`" and "`ip`" (also "`arp`") are also not available in the minimal "`debian" docker containers. 
 Which form the base-images we use in our 'code-server' teaching environments.  But then we also `--disable-telemetry`.

On the other hand to get the IP address this still worked...
perl -e 'printf "%s %s %s %s %vd\n",gethostbyname(shift)'  `hostname`
**host**.**domain**  **host** 2 4 10.0.0.186

**Comment by deansheather** [comment]
Moved the MAC stuff to a new issue: #911.

**Comment by fmder** [comment]
Looks like this is solved in v10.15.1.
